### DISCUSSION TITLE
Error while sending string parameters with especial characters like \n

### DISCUSSION THREAD

**Discussion by lujamaharjan** [fastapi/fastapi#9915] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

@app.post("/ep/")
async def endpoint1(input_data: str):
    #some coe
    return "hi"
```

### Description

I have an endpoint as shown above. When I call the above API using input_data that contains special characters such as \n. It gives me an unprocessable entity 422 error.

if I use a string like "hello world" it gives us 200.
But, If I send "hello \n world" it gives us a 422 status code.

### Operating System

Windows

### Operating System Details

in both windows and Linux

### FastAPI Version

0.79.0

### Python Version

3.7.6

**Comment by jgould22** [comment]
Can you share some examples of how you are calling it?

**Reply by lujamaharjan** [reply]
{
  "api_token": "",
  "mobile_number": "9868132399",
  "message": "sachin 
 test test"
}

**Comment by arjwilliams** [comment, marked as answer]
The way you are defining the endpoint in your example, FastAPI is treating input_data as a query parameter.

If you meant to pass the data as the body, you need to specify that:

```python
from fastapi import FastAPI, Body
...
@app.post("/ep/")
async def endpoint1(input_data: str = Body()):
    #some coe
    return "hi"
```

or specify the type as a Pydantic model:

```python
from pydantic import BaseModel
...
class MyModel(BaseModel):
    api_token: str
    mobile_number: str
    message: str

@app.post("/ep/")
async def endpoint1(input_data: MyModel):
    #some coe
    return "hi"
```

Note that FastAPI/Pydantic needs the body to be valid JSON in this case, and new lines within JSON strings is not valid JSON - new lines within strings need to be represented by "\n".

Invalid:
```json
{
    "key": "test
     value"
}
```
Valid:
```json
{
    "key": "test\nvalue"
}
```
